messagesigner/src/message.js

```diff
@@ -0,0 +1,212 @@
+/* eslint-disable valid-jsdoc */
+'use strict';
+
+var async = require('async');
+
+/*
+Factory function
+*/
+module.exports = function (ripplelib, sjcl) {
+
+var Remote = ripplelib.Remote;
+var Seed = ripplelib.Seed;
+var Account = ripplelib.Account;
+var UInt160 = ripplelib.UInt160;
+
+// Message class (static)
+var Message = {};
+
+Message.hashFunction = sjcl.hash.sha512.hash;
+Message.MAGIC_BYTES = 'Ripple Signed Message:\n';
+
+var REGEX_HEX = /^[0-9a-fA-F]+$/;
+var REGEX_BASE64 =
+  /^([A-Za-z0-9\+]{4})*([A-Za-z0-9\+]{2}==)|([A-Za-z0-9\+]{3}=)?$/;
+
+/**
+ *  Produce a Base64-encoded signature on the given message with
+ *  the string 'Ripple Signed Message:\n' prepended.
+ *
+ *  Note that this signature uses the signing function that includes
+ *  a recovery_factor to be able to extract the public key from the signature
+ *  without having to pass the public key along with the signature.
+ *
+ *  @static
+ *
+ *  @param {String} message
+ *  @param {sjcl.ecc.ecdsa.secretKey|Any format accepted by Seed.from_json}
+ *         secret_key
+ *  @param {RippleAddress} [The first key] account Field to specify the signing
+ *      account. If this is omitted the first account produced by the secret
+ *      generator will be used.
+ *  @return {Base64-encoded String} signature
+ */
+Message.signMessage = function(message, secret_key, account) {
+  return Message.signHash(Message.hashFunction(Message.MAGIC_BYTES + message),
+    secret_key, account);
+};
+
+/**
+ *  Produce a Base64-encoded signature on the given hex-encoded hash.
+ *
+ *  Note that this signature uses the signing function that includes
+ *  a recovery_factor to be able to extract the public key from the signature
+ *  without having to pass the public key along with the signature.
+ *
+ *  @static
+ *
+ *  @param {bitArray|Hex-encoded String} hash
+ *  @param {sjcl.ecc.ecdsa.secretKey|Any format accepted by Seed.from_json}
+ *          secret_key
+ *  @param {RippleAddress} [The first key] account Field to specify the
+ *          signing account. If this is omitted the first account produced by
+ *          the secret generator will be used.
+ *  @returns {Base64-encoded String} signature
+ */
+Message.signHash = function(_hash, secret_key_, account) {
+
+  var hash = typeof _hash === 'string' && /^[0-9a-fA-F]+$/.test(_hash) ?
+              sjcl.codec.hex.toBits(_hash) : _hash;
+
+  if (typeof hash !== 'object' ||
+      typeof hash[0] !== 'number' ||
+      hash.length <= 0) {
+    throw new Error('Hash must be a bitArray or hex-encoded string');
+  }
+
+  var secret_key = !(secret_key_ instanceof sjcl.ecc.ecdsa.secretKey) ?
+        Seed.from_json(secret_key_).get_key(account)._secret : secret_key_;
```

Comment:
This module is rendered pointless by relying upon Seed.get_key returning an object where there is an sjcl.bn `_secret` member available.

In newer versions of ripple-lib, using elliptic, this won't be the case. Given using this would necessitate an older version of ripple-lib which would itself likely have this functionality in it ...